server/tests/e2e/activities.e2e-spec.ts

```diff
@@ -0,0 +1,200 @@
+                                data: [
+                                    {
+                                        zoneNum: 0
+                                    },
+                                    {
+                                        zoneNum: 1
+                                    }
+                                ]
+                            }
+                        }
+                    }
+                },
+                credits: {
+                    create: {
+                        type: MapCreditType.AUTHOR,
+                        userID: user1.id
+                    }
+                },
+                images: {
+                    createMany: {
+                        data: [
+                            {
+                                small: 'bing.com',
+                                medium: 'bing.com',
+                                large: 'bing.com'
+                            },
+                            {
+                                small: 'bing.com',
+                                medium: 'bing.com',
+                                large: 'bing.com'
+                            }
+                        ]
+                    }
+                },
+                stats: {
+                    create: {
+                        reviews: 1
+                    }
+                }
+            },
+            include: {
+                info: true,
+                credits: true,
+                images: true,
+                stats: true,
+                tracks: {
+                    include: {
+                        zones: true
+                    }
+                }
+            }
+        })
+
+        activity1 = await prisma.activity.create({
+            data: {
+                userID: user1.id,
+                data: 122,
+                type: ActivityTypes.MAP_APPROVED
+            }
+        })
+
+        activity2 = await prisma.activity.create({
+            data: {
+                userID: user1.id,
+                data: 123,
+                type: ActivityTypes.WR_ACHIEVED
+            }
+        })
+
+        activity3 = await prisma.activity.create({
+            data: {
+                userID: user2.id,
+                data: 124,
+                type: ActivityTypes.REVIEW_MADE
+            }
+        })
+
+        const authService = global.auth as AuthService;
+        global.accessToken = (await authService.login(user1)).access_token;
+    });
+
+    describe('GET /api/v1/activities', () => {
+        const expects = (res: request.Response) => expect(res.body).toBeValidPagedDto(ActivityDto);
+        it('should respond with an array of activities', async () => {
+            const res = await get('activities', 200);
+
+            expects(res);
+            expect(res.body.totalCount).toBe(3);
+            expect(res.body.response[0].userID).toBe(user1.id);
+            expect(res.body.response[1].userID).toBe(user1.id);
```

Comment:
We can improve this test a little by also checking the `type` `data` values are right.

That being said, this test looks like it's going to depend on the order of activities being inserted. Also, if we're running Jest concurrently and there's some other test inserting activities, this data could be wrong.

I think a reasonable approach would be to first `.filter` the response data by say the user ID, then check the data and types are correct.  So how about

```ts
expect(res.body.totalCount).toBeGreaterThanOrEqual(3);
expect(res.body.response.find((data) => data.userID === user1.id && data.data === 122).type).toBe(ActivityTypes.MAP_APPROVED);
expect(res.body.response.find((data) => data.userID === user1.id && data.data === 123).type).toBe(ActivityTypes.WR_ACHIEVED);
expect(res.body.response.find((data) => data.userID === user2.id).type).toBe(ActivityTypes.REVIEW_MADE);
```